File: android/src/main/kotlin/com/apliarte/apliarte_flutter_connect/ApliarteFlutterConnectPlugin.kt
package com.apliarte.apliarte_flutter_connect

import android.os.Build
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** ApliarteFlutterConnectPlugin */
class ApliarteFlutterConnectPlugin : FlutterPlugin, MethodCallHandler {
    private var channel: MethodChannel? = null

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(
            flutterPluginBinding.binaryMessenger,
            "com.apliarte.flutter_connect/methods"
        )
        channel?.setMethodCallHandler(this)
    }

    override fun onMethodCall(call: MethodCall, result: Result) {
        when (call.method) {
            "initialize" -> result.success(null)
            "getCapabilities" -> result.success(
                mapOf(
                    "platformName" to "android-${Build.VERSION.SDK_INT}",
                    "supportsDiscovery" to false,
                    "supportsAdvertising" to false,
                    "supportsLocalNetworkTransport" to false,
                    "supportsPermissionRequests" to false,
                    "supportedRoles" to listOf("host", "viewer")
                )
            )
            "checkPermissions", "requestPermissions" -> result.success(
                mapOf(
                    "useCase" to (call.argument<String>("useCase") ?: "discoveryOnly"),
                    "status" to "unknown",
                    "requiredPermissions" to listOf("localNetwork"),
                    "missingPermissions" to listOf("localNetwork"),
                    "canRequest" to false,
                    "message" to "Native permission mapping is not implemented yet."
                )
            )
            "startDiscovery", "stopDiscovery", "dispose" -> result.success(null)
            else -> result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel?.setMethodCallHandler(null)
        channel = null
    }
}

File: ios/apliarte_flutter_connect/Sources/apliarte_flutter_connect/ApliarteFlutterConnectPlugin.swift
import Flutter
import UIKit

public class ApliarteFlutterConnectPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(
      name: "com.apliarte.flutter_connect/methods",
      binaryMessenger: registrar.messenger()
    )
    let instance = ApliarteFlutterConnectPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "initialize":
      result(nil)
    case "getCapabilities":
      result([
        "platformName": "ios-\(UIDevice.current.systemVersion)",
        "supportsDiscovery": false,
        "supportsAdvertising": false,
        "supportsLocalNetworkTransport": false,
        "supportsPermissionRequests": false,
        "supportedRoles": ["host", "viewer"]
      ])
    case "checkPermissions", "requestPermissions":
      let arguments = call.arguments as? [String: Any]
      result([
        "useCase": arguments?["useCase"] as? String ?? "discoveryOnly",
        "status": "unknown",
        "requiredPermissions": ["localNetwork"],
        "missingPermissions": ["localNetwork"],
        "canRequest": false,
        "message": "Native permission mapping is not implemented yet."
      ])
    case "startDiscovery", "stopDiscovery", "dispose":
      result(nil)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
